include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -3509,6 +3511,317 @@ namespace internal
+                  }
+            }
+        }
+
+      // sort for support points
+      std::sort(sp_to_l2_dofs.begin(),
+                sp_to_l2_dofs.end(),
+                [](const auto &a, const auto &b) { return a.first < b.first; });
+
+
+      // convert to CRS format
+      auto                                 it = sp_to_l2_dofs.begin();
+      std::vector<types::global_dof_index> l2_dofs;
+      l2_dofs.reserve(sp_to_l2_dofs.size());
+      std::vector<unsigned int> l2_dof_ptrs;
+      l2_dof_ptrs.reserve(dof_handler_h1.n_locally_owned_dofs() + 1);
+      l2_dof_ptrs.emplace_back(0);
+      for (unsigned int i = 0; i < dof_handler_h1.n_locally_owned_dofs(); ++i)
+        {
+          const unsigned int index = it->first;
+          while (it != sp_to_l2_dofs.end() && it->first == index)
+            {
+              l2_dofs.emplace_back(it->second);
+              ++it;
+            }
+          l2_dof_ptrs.emplace_back(l2_dofs.size());
+        }
+
+      return std::make_pair(std::move(l2_dof_ptrs), std::move(l2_dofs));
+    }
+
+
+    /**
+     * Same as above but a H1 dummy DoFHandler is created internally. Since it
+     * might be helpful to use the dummy DoFHandler outside the function it is
+     * returned in addition to the CRS structure.
+     *
+     * @param[in] dof_handler_l2 L2 conforming DoFHandler
+     * @param[in] constraint AffineConstrains associated with @p dof_handler_l2. Only unconstrained DoFs are considered
+     * @return A pair which contains
+     * 0) a pair of vectors which resembles a CRS structure to access global L2
+     * DoF indices from local support point (or H1 DoF) indices. 1) a shared
+     * pointer to the H1 conforming dummy DoFHandler created in the function
+     */
+    template <int dim, int spacedim, typename Number>
+    std::pair<std::pair<std::vector<unsigned int>,
+                        std::vector<types::global_dof_index>>,
+              std::shared_ptr<DoFHandler<dim, spacedim>>>
+    support_points_to_l2_dofs(
+      const DoFHandler<dim, spacedim> &        dof_handler_l2,
+      const dealii::AffineConstraints<Number> &constraint)
+    {
+      Assert(
+        dof_handler_l2.get_fe().degree > 0,
+        ExcMessage(
+          "Within this function a H1 dummy DoFHandler of the same degree as the L2 DoFHandler is built. H1 elements can not be of degree 0."));
+
+      Assert(dof_handler_l2.get_fe().conforming_space ==
+               FiniteElementData<dim>::Conformity::L2,
+             ExcMessage("dof_handler_l2 not L2 conforming"));
+
+      const auto &fe           = dof_handler_l2.get_fe();
+      const auto &tria         = dof_handler_l2.get_triangulation();
+      const auto  degree       = fe.degree;
+      const auto  n_components = fe.n_components();
+
+      // create h1 dummy dof handler
+      auto dof_handler_h1 = std::make_shared<DoFHandler<dim, spacedim>>(tria);
+      FESystem<dim, spacedim> fe_h1(FE_Q<dim, spacedim>(degree), n_components);
```

Comment:
Let's ignore multiple components for now.

Reply:
[f85dd34](https://github.com/dealii/dealii/pull/15209/commits/f85dd34bc167f7f13689909da7fda3bb07dd2ec4)